Apps/AdminWebClient/src/Server/Delegates/RestImmunizationAdminDelegate.cs

```diff
@@ -178,6 +186,24 @@ public async Task<RequestResult<PHSAResult<IList<ImmunizationViewResponse>>>> Ge
             return retVal;
         }
 
+        /// <summary>
+        /// Determines if the Immunization records are being refreshed and waits if true.
+        /// </summary>
+        /// <param name="result">The response from the Immunization delegate.</param>
+        /// <returns>true if a retry should be attempted by the delegate.</returns>
+        private bool Retry(RequestResult<PHSAResult<IList<ImmunizationViewResponse>>> result)
+        {
+            bool retry = result.ResultStatus == ResultType.Success &&
+                         result.ResourcePayload != null &&
+                         result.ResourcePayload.LoadState.RefreshInProgress;
+            if (retry)
+            {
+                Thread.Sleep(this.immunizationConfig.RetryWait);
+            }
```

Comment:
I think the sleep should be in the loop and this method only handle the check if it should retry to create a separation of concerns between the methods